client/js/components/image-viewer.js

```diff
@@ -0,0 +1,72 @@
+import { hasFullscreen, enterFullscreen, exitFullscreen } from '../util';
+import fastdom from '../utils/fastdom-promise';
+import { trackGaEvent } from '../tracking';
+import OpenSeadragon from 'openseadragon';
+
+function setupViewer(imageInfoSrc, viewer, viewerId) {
+  if (viewer.querySelector('.openseadragon-container')) return;
+  window.fetch(imageInfoSrc)
+  .then(function(response) {
+    return response.json();
+  })
+  .then((response) => {
+    OpenSeadragon({
+      id: `image-viewer-${viewerId}`,
+      visibilityRatio: 1,
+      showFullPageControl: false,
+      showHomeControl: false,
+      zoomInButton: `zoom-in-${viewerId}`,
+      zoomOutButton: `zoom-out-${viewerId}`,
+      showNavigator: true,
+      controlsFadeDelay: 0,
+      tileSources: [{
+        '@context': 'http://iiif.io/api/image/2/context.json',
+        '@id': response['@id'],
+        'height': response.height,
+        'width': response.width,
+        'profile': [ 'http://iiif.io/api/image/2/level2.json' ],
+        'protocol': 'http://iiif.io/api/image',
+        'tiles': [{
+          'scaleFactors': [ 1, 2, 4, 8, 16, 32 ],
+          'width': 400
+        }]
+      }]
+    });
+  }).catch(err => { throw err; });
+}
+
+const createImageViewer = (viewer) => {
+  if (window.fetch && hasFullscreen()) {
+    const image = viewer.previousElementSibling;
+    const viewerContent = viewer.querySelector('.image-viewer-fullscreen__content');
+    const viewerId = viewerContent.getAttribute('id');
+    const imageInfoSrc = document.getElementById(viewerId).getAttribute('data-info-src');
+    const enterFullscreenButton = viewer.querySelector('.js-enter-fullscreen');
+    const exitFullscreenButton = viewer.querySelector('.js-exit-fullscreen');
+
+    fastdom.mutate(() => {
+      viewer.style.display = 'block';
+    });
+
+    image.addEventListener('dblclick', (e) => {
+      setupViewer(imageInfoSrc, viewer, viewerId);
+      enterFullscreen(viewerContent);
+      trackGaEvent({
+        category: 'component',
+        action: 'work-enter-fullscreen-image:dblclick',
+        label: `id:${window.location.pathname.match(/\/works\/(.+)/)[1]}, title:${document.title}`
```

Comment:
Why are we storing the title in the analytics?
Could we not get this info from the component rather?